components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultOIDCClaimsCallbackHandler.java

```diff
@@ -241,18 +241,22 @@ private boolean isPreserverClaimUrisInAssertion(OAuthTokenReqMessageContext requ
     }
 
     private Map<ClaimMapping, String> getCachedUserAttributes(OAuthTokenReqMessageContext requestMsgCtx) {
-        Map<ClaimMapping, String> userAttributes = getUserAttributesCachedAgainstToken(getAccessToken(requestMsgCtx));
```

Comment:
Usually in the authorization code, once a token is issued the attributes cached against authorization code are stored against the token.

So ideal case, the same attributes stored against the authorization code will be available against the token at the point the token is id_token is built